1_slices_modules_and_objects/exercises/binary_search/src/lib.rs

```diff
@@ -1,7 +1,35 @@
 /// Finds the index of `value` in the sorted slice `v`
 /// Returns the index if found, None otherwise
 pub fn binary_search(v: &[i32], value: i32) -> Option<usize> {
-    unimplemented!();
+    let mut index = None;
+
+    if v.len() > 0 {
+
+        let mid: usize = v.len() / 2;
+
+
+        if v[mid] == value {
+            index = Some(mid);
+        }
+        else if v[mid] < value { 
+            index = binary_search(&v[mid+1..], value);           
+            match index {
+                Some(x) => index = Some(mid + x + 1),
+                None => (),
+            }
```

Comment:
This can be rewritten as
```
            index = binary_search(&v[mid+1..], value);           
            index = match index {
                Some(x) => Some(mid + x + 1),
                None => None,
            }
```
If you have a look at https://doc.rust-lang.org/std/option/enum.Option.html#method.map
You can see that map does exactly what this match does, meaning this can be simplified to
```suggestion
            index = binary_search(&v[mid+1..], value).map(|x| mid + x + 1);
```